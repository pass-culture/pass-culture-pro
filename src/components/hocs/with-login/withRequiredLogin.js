import { compose } from 'redux'
import withLogin from 'with-react-redux-login'

import { getRedirectToSignin } from './helpers'
import withFrenchQueryRouter from '../withFrenchQueryRouter'

const withRequiredLogin = compose(
  withFrenchQueryRouter,
  withLogin({
    handleFail: (state, action, { history, location }) =>
      history.push(getRedirectToSignin({...location})),
    isRequired: true,
  })
)

export default withRequiredLogin
