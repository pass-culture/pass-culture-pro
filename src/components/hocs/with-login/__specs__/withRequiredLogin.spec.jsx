import { mount } from 'enzyme'
import { createBrowserHistory } from 'history'
import React from 'react'
import { Provider } from 'react-redux'
import { Route, Router, Switch } from 'react-router-dom'

import {
  configureTestStore,
  configureFetchCurrentUserWithLoginFail,
} from './configure'
import OnMountCaller from './OnMountCaller'
import withRequiredLogin from '../withRequiredLogin'

const Test = () => null
const RequiredLoginTest = withRequiredLogin(Test)

describe('src | components | pages | hocs | with-login | withRequiredLogin', () => {
  afterEach(() => {
    fetch.resetMocks()
  })

  describe('functions', () => {
    it('should redirect to signin when not authenticated', async() => {await new Promise((resolve, reject) => {
      // given
      const history = createBrowserHistory()
      history.push('/test')
      const store = configureTestStore()
      configureFetchCurrentUserWithLoginFail()
      function onSuccessMountCallback() {
        expect(history.location.pathname).toStrictEqual("/connexion")
        resolve()
      }

      // when
      mount(
        <Provider store={store}>
          <Router history={history}>
            <Switch>
              <Route path="/test">
                <RequiredLoginTest />
              </Route>
              <Route path="/connexion">
                <OnMountCaller onMountCallback={onSuccessMountCallback} />
              </Route>
            </Switch>
          </Router>
        </Provider>
      )

      setTimeout(() => reject('Should have been redirected to /connexion'))
    })})
  })
})
