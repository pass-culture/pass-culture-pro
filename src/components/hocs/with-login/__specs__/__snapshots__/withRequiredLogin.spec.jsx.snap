// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | hocs | with-login | withRequiredLogin should match the snapshot 1`] = `ShallowWrapper {}`;
